crates/simulacrum/src/lib.rs

```diff
@@ -95,17 +97,21 @@ where
             .with_chain_start_timestamp_ms(1)
             .deterministic_committee_size(NonZeroUsize::new(1).unwrap())
             .build();
-        let keystore = KeyStore::from_newtork_config(&config);
+        Self::new_with_network_config(&config, rng)
+    }
+
+    pub fn new_with_network_config(config: &NetworkConfig, rng: R) -> Self {
```

Comment:
We likely do not want to allow someone to provide a network config as this essentially defeats the purpose of having deterministic generation of chain state.

Reply:
If the purpose of this is to allow for setting some initial distributions of accounts we likely should do so in a different way, perhaps by allowing someone to provide an account config.

Reply:
> We likely do not want to allow someone to provide a network config as this essentially defeats the purpose of having deterministic generation of chain state.

I disagree.I feel its okay to have a mode where one can come up with their own configs entirely and just use the engine to operate on that initial state. Whether or not their initial state is deterministic is up to them. The config builder provides an option for deterministic network config with the rng, so it should be fine.

Double-checking and the rng is only used for initial state in simulacrum anyways so I can probably remove it from the function sig and make it an optional in field in the `Simulacrum` struct.


> If the purpose of this is to allow for setting some initial distributions of accounts we likely should do so in a different way, perhaps by allowing someone to provide an account config.

For now I can add specific functions for what I know we want e.g initial genesis objects, and initial protocol config, but I expect more later. There are way too many things we would like to configure that one would end up replicating the network config builder for simulacrum.